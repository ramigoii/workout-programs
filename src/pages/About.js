import React from 'react';
import { FaUsers, FaGraduationCap, FaHandsHelping } from 'react-icons/fa';
import ourmission from '../styles/images/our-mission.jpg';

export const About = () => {
  return (
    <div className="about-container">
      <section className="about-hero">
        <div className="about-hero-content">
          <h1>О нас</h1>
          <p>Лучшие фитнес-программы для вашего здоровья и тела</p>
        </div>
      </section>

      <section className="about-content">
        <div className="about-text">
          <h2>Наша миссия</h2>
          <p>
            Мы создали платформу, которая поможет вам достичь ваших
            фитнес-целей. Наши программы разработаны профессионалами, чтобы
            обеспечить эффективные тренировки и мотивацию.
          </p>
          <p>
            Мы верим, что каждый человек заслуживает доступа к качественным
            тренировкам, независимо от уровня подготовки или местоположения.
            Наша цель — сделать фитнес доступным, эффективным и приятным для
            всех.
          </p>
          <button className="cta-button">Начать тренировки</button>
        </div>
        <div className="about-image">
          <img src={ourmission || '/placeholder.svg'} alt="Наша миссия" />
        </div>
      </section>

      <section className="about-values">
        <div className="container">
          <div className="section-title">
            <h2>Наши ценности</h2>
            <p>Принципы, которыми мы руководствуемся в нашей работе</p>
          </div>
          <div className="values-grid">
            <div className="value-item">
              <div className="value-icon">
                <FaUsers />
              </div>
              <h3>Фитнес для всех</h3>
              <p>
                Мы создаем программы, доступные каждому, независимо от уровня
                подготовки.
              </p>
            </div>
            <div className="value-item">
              <div className="value-icon">
                <FaGraduationCap />
              </div>
              <h3>Профессионализм</h3>
              <p>
                Все наши тренировки разработаны сертифицированными экспертами в
                области фитнеса.
              </p>
            </div>
            <div className="value-item">
              <div className="value-icon">
                <FaHandsHelping />
              </div>
              <h3>Сообщество</h3>
              <p>
                Мы создаем дружественную среду, где участники поддерживают и
                мотивируют друг друга.
              </p>
            </div>
          </div>
        </div>
      </section>
    </div>
  );
};

export default About;
