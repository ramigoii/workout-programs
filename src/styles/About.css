.about-container {
    font-family: 'Arial', sans-serif;
    color: #333;
    text-align: center;
  }
  
  .about-hero {
    background: url('./images/about-hero.jpg') no-repeat center center/cover;
    height: 50vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    color: white;
    text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.5);
  }
  
  .about-content {
    display: flex;
    flex-wrap: wrap;
    align-items: center;
    justify-content: space-around;
    padding: 50px 10%;
  }
  
  .about-text {
    max-width: 500px;
    text-align: left;
  }
  
  .about-image img {
    max-width: 250px;
    border-radius: 10px;
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.2);
  }
  
  .about-values {
    background: #f9f9f9;
    padding: 50px 0;
  }
  
  .values-grid {
    display: flex;
    flex-wrap: wrap;
    justify-content: center;
    gap: 30px;
  }
  
  .value-item {
    background: white;
    padding: 20px;
    border-radius: 10px;
    box-shadow: 0px 4px 8px rgba(0, 0, 0, 0.1);
    text-align: center;
    max-width: 250px;
  }
  
  .value-item img {
    width: 250px;
    margin-bottom: 10px;
  }
  
  @media (max-width: 768px) {
    .about-content {
      flex-direction: column;
      text-align: center;
    }
    .about-text {
      max-width: 100%;
    }
  }
  