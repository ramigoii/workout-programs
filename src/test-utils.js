import React from 'react';
import { createRoot } from 'react-dom/client';
import { act } from 'react-dom/test-utils';

// Simple render function
export function render(element) {
  const container = document.createElement('div');
  document.body.appendChild(container);
  const root = createRoot(container);
  
  act(() => {
    root.render(element);
  });
  
  return {
    container,
    getByText: (text) => {
      const elements = Array.from(container.querySelectorAll('*'))
        .filter(el => el.textContent === text);
      if (elements.length === 0) {
        throw new Error(`Could not find element with text: ${text}`);
      }
      return elements[0];
    },
    getByLabelText: (label) => {
      const labelElement = Array.from(container.querySelectorAll('label'))
        .find(el => el.textContent === label);
      if (!labelElement) {
        throw new Error(`Could not find label with text: ${label}`);
      }
      const input = document.getElementById(labelElement.htmlFor);
      if (!input) {
        throw new Error(`Could not find input for label: ${label}`);
      }
      return input;
    },
    getAllByTestId: (testId) => {
      return Array.from(container.querySelectorAll(`[data-testid^="${testId}"]`));
    },
    getByTestId: (testId) => {
      const element = container.querySelector(`[data-testid="${testId}"]`);
      if (!element) {
        throw new Error(`Could not find element with data-testid: ${testId}`);
      }
      return element;
    },
    unmount: () => {
      act(() => {
        root.unmount();
      });
      document.body.removeChild(container);
    }
  };
}

// Simple fireEvent utility
export const fireEvent = {
  click: (element) => {
    act(() => {
      const event = new MouseEvent('click', {
        bubbles: true,
        cancelable: true,
      });
      element.dispatchEvent(event);
    });
  },
  change: (element, { target }) => {
    act(() => {
      const event = new Event('change', {
        bubbles: true,
        cancelable: true,
      });
      Object.assign(element, target);
      element.dispatchEvent(event);
    });
  },
  submit: (element) => {
    act(() => {
      const event = new Event('submit', {
        bubbles: true,
        cancelable: true,
      });
      element.dispatchEvent(event);
    });
  }
};

// Simple expect utilities
export const expect = {
  toBeInTheDocument: (element) => {
    if (!document.body.contains(element)) {
      throw new Error('Element is not in the document');
    }
    return true;
  },
  toHaveTextContent: (element, text) => {
    if (element.textContent !== text) {
      throw new Error(`Expected element to have text content "${text}" but got "${element.textContent}"`);
    }
    return true;
  },
  toHaveValue: (element, value) => {
    if (element.value !== value) {
      throw new Error(`Expected element to have value "${value}" but got "${element.value}"`);
    }
    return true;
  }
};
export const mockPrograms = [
    {
      id: '1',
      title: 'Программа Сила',
      description: 'Описание силовой программы',
      schedule: [
        { day: 'Понедельник', exercises: ['Присед', 'Жим'] },
        { day: 'Среда', exercises: ['Становая тяга', 'Подтягивания'] },
      ],
    },
  ];
  
  export const mockProgramDetails = {
    '1': mockPrograms[0],
  };
  