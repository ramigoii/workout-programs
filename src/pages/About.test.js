import React from 'react';
import { render, screen } from '@testing-library/react';
import { About } from './About';
import { act } from 'react-dom/test-utils';

test('About page renders content', async () => {
  await act(async () => {
    render(<About />);
  });

  expect(screen.getByText('О нас')).toBeInTheDocument();
  expect(screen.getByText('Наша миссия')).toBeInTheDocument();
  expect(screen.getByText('Наши ценности')).toBeInTheDocument();

  // Проверка пунктов ценностей
  expect(screen.getByText('Фитнес для всех')).toBeInTheDocument();
  expect(screen.getByText('Профессионализм')).toBeInTheDocument();
  expect(screen.getByText('Сообщество')).toBeInTheDocument();
});

test('matches snapshot', async () => {
  let container;
  await act(async () => {
    const rendered = render(<About />);
    container = rendered.container;
  });
  expect(container).toMatchSnapshot();
});
