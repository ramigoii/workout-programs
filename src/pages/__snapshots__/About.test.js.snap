// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div>
  <div
    class="about-container"
  >
    <section
      class="about-hero"
    >
      <div
        class="about-hero-content"
      >
        <h1>
          О нас
        </h1>
        <p>
          Лучшие фитнес-программы для вашего здоровья и тела
        </p>
      </div>
    </section>
    <section
      class="about-content"
    >
      <div
        class="about-text"
      >
        <h2>
          Наша миссия
        </h2>
        <p>
          Мы создали платформу, которая поможет вам достичь ваших фитнес-целей. Наши программы разработаны профессионалами, чтобы обеспечить эффективные тренировки и мотивацию.
        </p>
        <p>
          Мы верим, что каждый человек заслуживает доступа к качественным тренировкам, независимо от уровня подготовки или местоположения. Наша цель — сделать фитнес доступным, эффективным и приятным для всех.
        </p>
        <button
          class="cta-button"
        >
          Начать тренировки
        </button>
      </div>
      <div
        class="about-image"
      >
        <img
          alt="Наша миссия"
          src="our-mission.jpg"
        />
      </div>
    </section>
    <section
      class="about-values"
    >
      <div
        class="container"
      >
        <div
          class="section-title"
        >
          <h2>
            Наши ценности
          </h2>
          <p>
            Принципы, которыми мы руководствуемся в нашей работе
          </p>
        </div>
        <div
          class="values-grid"
        >
          <div
            class="value-item"
          >
            <div
              class="value-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 224c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm448 0c35.3 0 64-28.7 64-64s-28.7-64-64-64-64 28.7-64 64 28.7 64 64 64zm32 32h-64c-17.6 0-33.5 7.1-45.1 18.6 40.3 22.1 68.9 62 75.1 109.4h66c17.7 0 32-14.3 32-32v-32c0-35.3-28.7-64-64-64zm-256 0c61.9 0 112-50.1 112-112S381.9 32 320 32 208 82.1 208 144s50.1 112 112 112zm76.8 32h-8.3c-20.8 10-43.9 16-68.5 16s-47.6-6-68.5-16h-8.3C179.6 288 128 339.6 128 403.2V432c0 26.5 21.5 48 48 48h288c26.5 0 48-21.5 48-48v-28.8c0-63.6-51.6-115.2-115.2-115.2zm-223.7-13.4C161.5 263.1 145.6 256 128 256H64c-35.3 0-64 28.7-64 64v32c0 17.7 14.3 32 32 32h65.9c6.3-47.4 34.9-87.3 75.2-109.4z"
                />
              </svg>
            </div>
            <h3>
              Фитнес для всех
            </h3>
            <p>
              Мы создаем программы, доступные каждому, независимо от уровня подготовки.
            </p>
          </div>
          <div
            class="value-item"
          >
            <div
              class="value-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M622.34 153.2L343.4 67.5c-15.2-4.67-31.6-4.67-46.79 0L17.66 153.2c-23.54 7.23-23.54 38.36 0 45.59l48.63 14.94c-10.67 13.19-17.23 29.28-17.88 46.9C38.78 266.15 32 276.11 32 288c0 10.78 5.68 19.85 13.86 25.65L20.33 428.53C18.11 438.52 25.71 448 35.94 448h56.11c10.24 0 17.84-9.48 15.62-19.47L82.14 313.65C90.32 307.85 96 298.78 96 288c0-11.57-6.47-21.25-15.66-26.87.76-15.02 8.44-28.3 20.69-36.72L296.6 284.5c9.06 2.78 26.44 6.25 46.79 0l278.95-85.7c23.55-7.24 23.55-38.36 0-45.6zM352.79 315.09c-28.53 8.76-52.84 3.92-65.59 0l-145.02-44.55L128 384c0 35.35 85.96 64 192 64s192-28.65 192-64l-14.18-113.47-145.03 44.56z"
                />
              </svg>
            </div>
            <h3>
              Профессионализм
            </h3>
            <p>
              Все наши тренировки разработаны сертифицированными экспертами в области фитнеса.
            </p>
          </div>
          <div
            class="value-item"
          >
            <div
              class="value-icon"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 640 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M488 192H336v56c0 39.7-32.3 72-72 72s-72-32.3-72-72V126.4l-64.9 39C107.8 176.9 96 197.8 96 220.2v47.3l-80 46.2C.7 322.5-4.6 342.1 4.3 357.4l80 138.6c8.8 15.3 28.4 20.5 43.7 11.7L231.4 448H368c35.3 0 64-28.7 64-64h16c17.7 0 32-14.3 32-32v-64h8c13.3 0 24-10.7 24-24v-48c0-13.3-10.7-24-24-24zm147.7-37.4L555.7 16C546.9.7 527.3-4.5 512 4.3L408.6 64H306.4c-12 0-23.7 3.4-33.9 9.7L239 94.6c-9.4 5.8-15 16.1-15 27.1V248c0 22.1 17.9 40 40 40s40-17.9 40-40v-88h184c30.9 0 56 25.1 56 56v28.5l80-46.2c15.3-8.9 20.5-28.4 11.7-43.7z"
                />
              </svg>
            </div>
            <h3>
              Сообщество
            </h3>
            <p>
              Мы создаем дружественную среду, где участники поддерживают и мотивируют друг друга.
            </p>
          </div>
        </div>
      </div>
    </section>
  </div>
</div>
`;
